connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -562,7 +556,6 @@ public boolean startTask(
                 // statusListener
                 Plugins.compareAndSwapLoaders(savedLoader);
                 connectorStatusMetricsGroup.recordTaskRemoved(id);
-                workerMetricsGroup.recordTaskFailure();
```

Comment:
I think we might want to keep this line here in case we fail somehow before even creating the `WorkerTask` instance. This can happen if a `Converter`, `Transformation`, etc. throws an exception during startup.

Reply:
Good point. I believe we could achieve the same effect by wrapping the statusListener earlier in the method and using the wrapped listener in the exception handler. That way the only way we could fail without recording a metric would be if the wrapping itself failed (which is a dead simple constructor, so seems unlikely)

Reply:
Sounds good!